caffeine/src/main/java/com/github/benmanes/caffeine/cache/AsyncCache.java

```diff
@@ -170,7 +181,7 @@ CompletableFuture<Map<K, V>> getAll(Iterable<? extends K> keys,
    * @param valueFuture the value to be associated with the specified key
    * @throws NullPointerException if the specified key or value is null
    */
-  void put(K key, CompletableFuture<? extends V> valueFuture);
+  void put(K key, CompletableFuture<? extends @Nullable V> valueFuture);
```

Comment:
This is basically unrelated. I just noticed that it's another case in which null might actually be valid, and it's a case in which there's no _return type_ that we might choose to _also_ make nullable.

Reply:
Resolving based on the thumbs-up emoji.